docs/crds/hostrule.md

```diff
@@ -309,6 +309,105 @@ The `networkSecurityPolicy` setting, in addition to any other parameters provide
+                    name: my-app-1
+                    port:
+                      number: 80
+                path: /bar/foo
+                pathType: ImplementationSpecific
+          tls:
+          - hosts:
+            - test-avi.vmware.com
+            secretName: ingress-secret
+
+   For the above Ingress definition, if we create a hostrule with `useRegex` set to True, then :  
+   a) A request that comes on the URL **https://test-avi.vmware.com/bar/foo** will be routed to backend **my-app-1**.  
+   b) A request that comes on the URL **https://test-avi.vmware.com/bar/baz** will be routed to backend **my-app-0**.
+
+2. If a hostrule is defined with the useRegex field set to **False**, or no hostrule is defined for a host and the user specifies paths that are regular expressions in Ingress/Route for the same host, then Avi will not consider the paths as regular expressions and will create rules that will match the whole path instead of regex pattern matching.
+
+***Note***
+1. This property is available only in HostRule `v1beta1` schema definition.
+2. The `useRegex` field will not be supported for insecure ingress/route when AKO is running in SNI mode or SNI Dedicated mode.
+3. For secure SNI and secure SNI dedicated virtual services, pool creation may fail in Avi if regular expression paths are defined and hostrule is not defined with **useRegex** set to **True**. This limitation is because the naming convention for pools followed by AKO includes path in the pool name while some special characters are not allowed in pool names by Avi Controller.
+4. To overcome the above limitation for secure SNI and secure SNI dedicated virtual services, the user should always define a hostrule with **useRegex** set to **True** if regular expression paths are defined in secure Ingress/Route. If **useRegex** is **True**, AKO will create the pool with an encoded name.
+
+#### Express Application Root Redirect path
+If the application root path, i.e., path `/`, for a given host, defined in Ingress or Route, is exposed on a different path in the backend and needs to be redirected, then set this field to the required path in a hostrule to redirect requests. This is an optional field. The host and query strings in the redirect URI will be kept the same. The port in the redirect request will be set as the front-end port specified in the ingress, and the redirect protocol will be set based on the redirect port (HTTPS for ports 443 and 6443 and HTTP for any other port).
+
+        fqdn: test-avi.vmware.com
+        fqdnType: Exact
+        applicationRootPath: "/foo"
+
+If any other path (along with the application root **/**) is also specified for the host in the ingress/route, then requests for that path will be routed normally (without any redirect) by Avi. The path that is used as the **applicationRootPath** should not also be used as a regular path in the ingress/route, as it will lead to ambiguity while backend selection in Avi.
```

Comment:
Should be rephrased as `If any other path (along with the application root **/**) is also specified for the host in the ingress/route, then Avi will route requests for that path normally (without any redirect). The path used as the **applicationRootPath** should not be used as a regular path in the Ingress/Route, as it will lead to ambiguity during backend selection in Avi.`

Reply:
Both changes done.